modules/lib/lib-event/src/main/resources/lib/xp/event.ts

```diff
@@ -0,0 +1,136 @@
+import * as Lib from './event';
+
+/*
+ * Global library definitions for require and __non_webpack_import__
+ */
+
+type LibType = typeof Lib;
+
+declare global {
+    interface XpEventLibrary extends LibType {}
+}
+
+
+export interface ListenerParams<EventData extends object = EnonicEventData> {
+    /**
+     * Event type pattern
+     */
+    type: EnonicEventTypes;
+
+    /**
+     * Callback event listener
+     */
+    callback: (event: EnonicEvent<EventData>) => void;
+
+    /**
+     * Local events only (default to false)
+     */
+    localOnly?: boolean;
+}
+
+export interface SendParams {
+    /**
+     * Event type
+     */
+    type: EnonicEventTypes;
+
+    /**
+     * `true` if it should be distributed in cluster
+     */
+    distributed?: boolean;
+
+    /**
+     * Additional data for event.
+     */
+    data?: object;
+}
+
+export interface EnonicEvent<EventData extends object = EnonicEventData> {
+    readonly type: EnonicEventTypes | string;
```

Comment:
Make sure that `readonly` is actually what you want for these cases.

Reply:
Sure, will look. BTW, this part is basically the copy from `enonic-types`, so I haven't dig into it yet.

Reply:
This is what I expected. When I started *enonic-types* I used a lot of `readonly`, because my projects use a lot of "defensive copying" (no mutation of objects).

Since this isn't everybodys style, I have removed a lot of `readonly` since then.

But since you guys are starting a bit fresh, it would be worth creating some "rules" for where you wish to use it.